### PR TITLE
Render “respond-to/confirm offer” page in modal

### PR DESCRIPTION
**Pull request by lilyfromseattle**
# [PURCHASE-2712]

As a collector I want to be presented with the contents of “respond-to/confirm offer“ page in modal form when clicking on the CTA to follow these actions.

Acceptance criteria:
- ”Respond-to/confirm offer“ page should render in modal.

- CTA should trigger modal on ”respond-to/confirm offer“ state.

I wasn't sure what tests were needed for this work, as the modal only contains an iframe, but if anyone has any suggestions I'm open!

Screen Shot 2021-06-22 at 3 26 17 PM

[PURCHASE-2712]: https://artsyproduct.atlassian.net/browse/PURCHASE-2712

### CODE REVIEW

**Comment by damassi** [comment]
Nice work @lilyfromseattle 💯 

To confirm, once the modal pops up you're able to complete the full order flow? What happens at the end of the last step?

File: src/v2/Apps/Conversation/Components/ReviewOfferCTA.tsx [hunk 1]
```diff
@@ -132,54 +118,47 @@ export const ReviewOfferCTA: React.FC<ReviewOfferCTAProps> = ({
...
+    <>
+      <ClickableFlex
+        px={2}
+        py={1}
+        justifyContent="space-between"
+        alignItems="center"
+        bg={backgroundColor}
+        flexDirection="row"
+        minHeight={60}
+        onClick={() => {
```
Comment (xtina-starr) [inline src/v2/Apps/Conversation/Components/ReviewOfferCTA.tsx:143]: You mentioned about test in your PR description. Maybe you could simulate this click the check if `<StyledIframe>`  is present?

File: src/v2/Apps/Conversation/Utils/returnOrderModalDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { CommerceBuyerOfferActionEnum } from "v2/__generated__/ConversationCTA_conversation.graphql"
+
+export const returnOrderModalDetails = ({
```
Comment (xtina-starr) [inline src/v2/Apps/Conversation/Utils/returnOrderModalDetails.tsx:3]: Nice helper function!

File: src/v2/Apps/Order/OrderApp.tsx [hunk 3]
```diff
@@ -135,18 +135,22 @@ class OrderApp extends React.Component<OrderAppProps, {}> {
...
           // @ts-expect-error STRICT_NULL_CHECK
           this.mediator = mediator
+
           return (
             <Box>
               {/* FIXME: remove once we refactor out legacy backbone code.
-                  Add place to attach legacy flash message, used in legacy inquiry flow
-              */}
+                    Add place to attach legacy flash message, used in legacy inquiry flow
+                */}
```
Comment (xtina-starr) [inline src/v2/Apps/Order/OrderApp.tsx:151]: Is this comment and`<div id="main-layout-flash" />` still needed?

File: src/v2/Apps/Order/OrderApp.tsx [hunk 4]
```diff
@@ -135,18 +135,22 @@ class OrderApp extends React.Component<OrderAppProps, {}> {
...
               <div id="main-layout-flash" />
-              <MinimalNavBar to={artworkHref}>
+              <MinimalNavBar to={artworkHref} isBlank={isModal}>
```
Comment (xtina-starr) [inline src/v2/Apps/Order/OrderApp.tsx:153]: What does `isBlank` refer to here? What is the use case for `isBlank`?

Comment (lilyfromseattle) [reply src/v2/Apps/Order/OrderApp.tsx:153]: `isBlank` hides the logo that is usually present in `MinimalNavBar`. `MinimalNavBar` suppresses the default nav bar which we want, we just don't want any content there at all

File: src/v2/Apps/Order/OrderApp.tsx (hunk 3 above)
Comment (lilyfromseattle) [reply src/v2/Apps/Order/OrderApp.tsx:151]: We're still using the legacy inquiry flow so I think it has to stay put for now

**Comment by lilyfromseattle** [comment]
@damassi Just saw this! Yes, once you submit the counteroffer it takes you to the status page saying your offer was submitted and you can close the modal from there

File: src/v2/Apps/Order/OrderApp.tsx (hunk 3 above)
Comment (lilyfromseattle) [reply src/v2/Apps/Order/OrderApp.tsx:151]: We still use the legacy inquiry flow, so I think we should probably leave it in for now

File: src/v2/Apps/Conversation/Components/__tests__/ConversationCTA.jest.tsx [hunk 5]
```diff
@@ -159,4 +160,19 @@ describe("ConversationCTA", () => {
 
     expect(wrapper.find("ReviewOfferCTA").length).toBe(0)
   })
+
+  it("clicking the review offer CTA opens the order modal", () => {
```
Comment (xtina-starr) [inline src/v2/Apps/Conversation/Components/__tests__/ConversationCTA.jest.tsx:164]: 🔥 🔥 🔥

**Pull request merged**